src/System.Runtime/tests/System/GC.cs

```diff
@@ -357,4 +358,26 @@ public static void TestGetGeneration()
             GC.Collect();
         }
     }
+
+    [Fact]
+    public static void TestLargeObjectHeapCompactionModeRoundTrips()
+    {
+        var before = GCSettings.LargeObjectHeapCompactionMode;
+        var different = before == GCLargeObjectHeapCompactionMode.CompactOnce ? GCLargeObjectHeapCompactionMode.Default : GCLargeObjectHeapCompactionMode.CompactOnce;
+        GCSettings.LargeObjectHeapCompactionMode = different;
+        Assert.Equal(different, GCSettings.LargeObjectHeapCompactionMode);
+        GCSettings.LargeObjectHeapCompactionMode = before;
+        Assert.Equal(before, GCSettings.LargeObjectHeapCompactionMode);
```

Comment:
You might consider changing this to something like:
```C#
[Theory]
[InlineData(GCLargeObjectHeapCompactionMode.CompactOnce)]
[InlineData(GCLargeObjectHeapCompactionMode.Default)]
public static void TestLargeObjectHeapCompactionModeRoundTrips(GCLargeObjectHeapCompactionMode value)
{
    GCLargeObjectHeapCompactionMode orig = GCSettings.LargeObjectHeapCompactionMode;
    try
    {
        GCSettings.LargeObjectHeapCompactionMode = value;
        Assert.Equal(value, GCSettings.LargeObjectHeapCompactionMode);
    }
    finally
    {
        GCSettings.LargeObjectHeapCompactionMode = orig;
        Assert.Equal(orig, GCSettings.LargeObjectHeapCompactionMode);
    }
}
```
That'll test setting both values, will ensure it gets reset even if there's an exception / an assert fails, and avoids the complexity of having to switch on the input to determine the other value.

Same applies to the test below.